src/Utils/ObjectMixin.php

```diff
@@ -25,14 +27,8 @@ class ObjectMixin
 	/** @var array (name => [type => callback])  used by get|setExtensionMethod() */
 	private static $extMethods = [];
 
-
-	/**
-	 * Static class - cannot be instantiated.
-	 */
-	final public function __construct()
-	{
-		throw new Nette\StaticClassException;
-	}
+	/** @var array (name => [read name, write?, returnsByRef?])  used by getMagicProperties() */
+	private static $properties = [];
```

Comment:
How would you implement events without `$props`? You can rename in the future the property to `$events` but you still need to store somewhere the information from reflection check.